### PR TITLE
[Stdlib] Add tests for removeprefix() and removesuffix() on String and StringSlice

### PR DESCRIPTION
**Pull request by msaelices**
Expand test coverage for `removeprefix()` and `removesuffix()`:

- Add `test_removeprefix()` and `test_removesuffix()` to `test_string_slice.mojo` covering `StringSlice` behavior.
- Expand existing tests in `test_string.mojo` to cover `String` (previously only `StaticString` was tested).

New cases: prefix/suffix longer than string, only-removes-once semantics, and Unicode strings.

---

Assisted-by: AI

### CODE REVIEW

File: mojo/stdlib/test/collections/string/test_string_span.mojo [hunk 1]
```diff
@@ -881,6 +881,56 @@ def test_endswith() raises:
...
+    # Middle substring is not treated as prefix.
+    assert_equal(
+        StringSlice("hello world").removeprefix("llo wor"), "hello world"
+    )
+    # Only removes once, not recursively.
+    assert_equal(StringSlice("aahello").removeprefix("aa"), "hello")
+    assert_equal(StringSlice("aaaahello").removeprefix("aa"), "aahello")
+    # Unicode prefix.
+    assert_equal(StringSlice("héllo").removeprefix("hé"), "llo")
+    assert_equal(StringSlice("héllo").removeprefix("ll"), "héllo")
```
Comment (JoeLoser) [inline mojo/stdlib/test/collections/string/test_string_span.mojo:957]: **Suggestion** Consider adding an empty-on-empty boundary case and more multi-byte Unicode coverage:

```mojo
    # Empty string with empty prefix.
    assert_equal(StringSlice("").removeprefix(""), "")
    # Emoji prefix.
    assert_equal(StringSlice("hello🌍").removeprefix("hello"), "🌍")
    assert_equal(StringSlice("🌍hello").removeprefix("🌍"), "hello")
```

(And the equivalent for `test_removesuffix` below.)

The emoji cases exercise multi-byte codepoints at the boundary, which helps catch any naive byte-level slicing.

File: mojo/stdlib/test/collections/string/test_string.mojo [hunk 2]
```diff
@@ -1061,6 +1061,22 @@ def test_removeprefix() raises:
...
+    # Only removes once, not recursively.
+    var s3 = String("aaaahello")
+    assert_equal(s3.removeprefix("aa"), "aahello")
+    # Unicode prefix.
+    var s4 = String("héllo")
+    assert_equal(s4.removeprefix("hé"), "llo")
+    assert_equal(s4.removeprefix("ll"), "héllo")
 
 
 def test_removesuffix() raises:
```
Comment (JoeLoser) [inline mojo/stdlib/test/collections/string/test_string.mojo:1105]: **Suggestion** Same suggestion here — consider adding `assert_equal(String("").removeprefix(""), "")` for the empty-on-empty edge case, plus an emoji test like `String("🌍hello").removeprefix("🌍")` to exercise multi-byte prefix removal on `String`.

**Review by JoeLoser — APPROVED** [review]
LGTM % minor suggestions. Please take a look and then I'm happy to sync this in afterwards. Thank you for the contribution!

**Comment by msaelices** [comment]
@JoeLoser rebased onto latest `main` and it's mergeable again. This is test-only coverage you approved earlier. Happy to have it `!sync`'d whenever convenient. Thanks!
